src/main/java/org/broadinstitute/hellbender/tools/ApplyBQSR.java

```diff
@@ -54,7 +54,7 @@
     @Override
     public void onTraversalStart() {
         final SAMFileHeader outputHeader = ReadUtils.cloneSAMFileHeader(getHeaderForReads());
```

Comment:
Once you fix `createSAMWriter()` so that you don't have to manually pass in a header and reference, delete these `outputHeader` creation lines duplicated in all the tools' `onTraversalStart()` methods.